master/docs/developer/style.rst

```diff
@@ -122,6 +123,15 @@ Deferred.  As a shortcut, ``d.addCallback`` works as a decorator::
     def rev_parse(_): # note use of '_' to quietly indicate an ignored parameter
         return utils.getProcessOutput(git, [ 'rev-parse', revname ])
 
+.. note::
+
+    ``d.addCallback`` is not really a decorator as it does not return a modified function.
+    As a result in previous code, ``rev_parse`` value is actually the Deferred.
+    In general the :class:`inlineCallbacks` method is preferred inside new code as it keeps the code easier to read.
```

Comment:
It does not this is just used for styling. We don't have back reference helpers to twisted docs